beacon-chain/rpc/beacon/assignments.go

```diff
@@ -178,38 +188,46 @@ func (bs *Server) NextEpochProposerList(
 	}, nil
 }
 
-// prepareProposerList
-func prepareProposerAssignments(curEpochState, nextEpochState *stateTrie.BeaconState,
-	curEpoch, nextEpoch types.Epoch)  ([]*ethpb.ValidatorAssignments_CommitteeAssignment, error) {
+// GetProposerList retrieves the validator assignments for a given epoch, [This api is specially used for Orchestrator client]
+// optional validator indices or public keys may be included to filter validator assignments.
+func (bs *Server) NextEpochProposerList(
+	ctx context.Context,
+	empty *ptypes.Empty,
+) (assignments *ethpb.ValidatorAssignments, err error) {
+	curEpoch := helpers.SlotToEpoch(bs.GenesisTimeFetcher.CurrentSlot())
+	assignments, err = bs.GetProposerListForEpoch(ctx, curEpoch)
 
-	var res []*ethpb.ValidatorAssignments_CommitteeAssignment
-	// Initialize all committee related data.
-	proposerListCurEpoch, err := helpers.ProposerAssignments(curEpochState, curEpoch)
+	return
+}
+
+func (bs *Server) GetMinimalConsensusInfo(
```

Comment:
Why do we need this `GetMinimalConsensusInfo` now? We do not call this api now.

Reply:
It's called by api backend of the subscriber rpc implementation: `consensusInfo, currentErr := beaconChain.GetMinimalConsensusInfo(ctx, blockEpoch)`